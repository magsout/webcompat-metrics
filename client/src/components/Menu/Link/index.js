/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import PropTypes from "prop-types";
import { Link as LinkRouter } from "react-router-dom";

import classes from "./styles.css";

const Link = ({ to, text, icon }) => {
  return (
    <LinkRouter to={to}>
      <div className={classes.component}>
        {icon && <div className={classes.icon}>{icon}</div>}
        {text && <div className={classes.text}>{text}</div>}
      </div>
    </LinkRouter>
  );
};

Link.propTypes = {
  to: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  icon: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
};

Link.defaultProps = {
  text: null,
  icon: null,
};

export default Link;
