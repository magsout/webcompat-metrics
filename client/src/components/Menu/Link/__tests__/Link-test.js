/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

import Link from "../";

it("renders Link default correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <Link to="/some/route" />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Link with text and icon correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <Link to="/some/route" text={"some text"} icon={"some icon"} />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Link with node text / icon correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <Link
          to="/some/route"
          text={<span>some node text</span>}
          icon={<span>some node icon</span>}
        />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
