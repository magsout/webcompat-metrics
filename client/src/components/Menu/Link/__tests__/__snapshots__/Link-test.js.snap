// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Link default correctly 1`] = `
<a
  href="/some/route"
  onClick={[Function]}
>
  <div
    className="component"
  />
</a>
`;

exports[`renders Link with node text / icon correctly 1`] = `
<a
  href="/some/route"
  onClick={[Function]}
>
  <div
    className="component"
  >
    <div
      className="icon"
    >
      <span>
        some node icon
      </span>
    </div>
    <div
      className="text"
    >
      <span>
        some node text
      </span>
    </div>
  </div>
</a>
`;

exports[`renders Link with text and icon correctly 1`] = `
<a
  href="/some/route"
  onClick={[Function]}
>
  <div
    className="component"
  >
    <div
      className="icon"
    >
      some icon
    </div>
    <div
      className="text"
    >
      some text
    </div>
  </div>
</a>
`;
